pkgs/development/libraries/dnnl/default.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchFromGitHub, cmake, substituteAll, bash, autoPatchelfHook }:
+
+stdenv.mkDerivation rec {
+  pname = "dnnl";
+  version = "1.0.2";
+
+  src = fetchFromGitHub {
+    owner = "intel";
+    repo = "mkl-dnn";
+    rev = "v${version}";
+    sha256 = "0lf14a4j511pydyz78wwbl6hifbq86m3dy119iss9cvz8zzrdyvw";
+  };
+  outputs = [ "out" "dev" "doc" ];
+  patches = [ (substituteAll {
+    src = ./bash-to-sh.patch;
+    inherit bash;
+  }) ];
+
+  nativeBuildInputs = [ cmake ];
+
+  doCheck = true;
+
+  # the test executables aren't able to find libmkldnn.so through LD_LIBRARY_PATH, so patch them.
+  checkInputs = [ autoPatchelfHook ];
+  preCheck = ''
+    addAutoPatchelfSearchPath $PWD/src
+    autoPatchelf tests examples
+  '';
+
```

Comment:
This doesn't seem to work

Reply:
The checkPhase runs after build, but before install, so $out should still be empty when this gets called

Reply:
Another options would be `export LD_LIBRARY_PATH=$PWD/src:$LD_LIBRARY_PATH`. When i tried originally, i left off the export, and that seems to make a difference in some cases.

Reply:
perhaps:
```
  checkPhase = ''
    LD_LIBRARY_PATH=$PWD/src:$LD_LIBRARY_PATH make check
  '';
```
would work best

Reply:
This works, is this preferable?

Reply:
I honestly have no idea

Reply:
What you're doing seems to work, I was just hoping to simplify the rule at the cmake level.

But looking further at this the RPATH on these tests clearly are wrong.  I also couldn't find a nice way to override this for since the 1.0 release.

I think either your original or this proposed solution work in terms of actual functionality and sorry for sidetracking.

Reply:
strangely I didn't need to do anything when I built it in a nix-shell

Reply:
if you did `nix-shell --pure` then that might not be the case.

Reply:
even then it works, I can't work it out. I ever messed around to get it to run tree in the checkphase and all the files look identical

Reply:
well, if it works, then it works. I just remember getting the "libdnn.so" cannot be found error

Reply:
In a nutshell, I believe the behavior is broken because of CMAKE_SKIP_BUILD_RPATH=ON
(Nix open issue and workarounds here: https://github.com/NixOS/nixpkgs/issues/22060).

I think you can resolve this with your solution or adopt one of the others from that thread.